### PR TITLE
[Lang] Define __str__ method for DataType

### LINKED ISSUE #3367: Properly support printing taichi data type string
I'd expect this to print `ti.u16` instead of the object info
```
>>> str(ti.u16)
'<taichi.lib.taichi_core.DataType object at 0x7f8af4a9b4f0>'
```
This will make composing error message string much easier. 
(note this information can be retrieved by `ti.u16.to_string()` but this is not very pythonic. :P

### PR DESCRIPTION
**Pull request by knight42**
Signed-off-by: knight42 <anonymousknight96@gmail.com>
fixes #3367

### CODE REVIEW

File: taichi/python/export_lang.cpp [hunk 1]
```diff
@@ -103,6 +103,7 @@ void export_lang(py::module &m) {
       .def(py::self == py::self)
       .def("__hash__", &DataType::hash)
       .def("to_string", &DataType::to_string)
+      .def("__str__", &DataType::to_string)
```
Comment (yihong0618) [inline taichi/python/export_lang.cpp:106]: do we need add `__repr__` too?

**Review by yihong0618 — APPROVED** [review]

File: taichi/python/export_lang.cpp (hunk 1 above)
Comment (knight42) [reply taichi/python/export_lang.cpp:106]: SGTM!

Comment (ailzhang) [reply taichi/python/export_lang.cpp:106]: `__repr__` is more like for debugging purpose which object information. 
I usually think in this way (maybe incorrect :P):
information from `__str__` should be human readable (but can lose info)
But you can use information from `__repr__` to reconstruct this object easily. 
So IMHO only handling `__str__` is good enough for now and we might consider adding `__repr__` in the future :D

**Review by ailzhang — APPROVED** [review]
Nice thanks!

**Review by ailzhang — APPROVED** [review]
Thanks!!

**Comment by yihong0618** [comment]
cool

**Pull request merged**